lib/web_catalog/object_graph_importer.es6.js

```diff
@@ -134,34 +134,65 @@ foam.CLASS({
           if (stat.isFile()) {
             if (!ogFileRegExp.test(objectGraphFiles[i])) continue;
 
-            // Drop slice(0, -5): ".json".
-            let releaseInfo = objectGraphFiles[i].slice(0, -5).split('_');
-            let bName = releaseInfo[1];
-            let bVersion = releaseInfo[2];
-            let osName = releaseInfo[3];
-            let osVersion = releaseInfo[4];
+            let releaseInfo = this.getReleaseInfo(objectGraphFiles[i]);
 
             this.info(`Extracting APIs from
-                ${bName} ${bVersion} ${osName} ${osVersion}`);
+                ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                ${releaseInfo.platform.name} ${releaseInfo.platform.version}`);
             promises.push(
                 this.apiExtractorService.extractWebCatalog(filePath)
                 .then(catalog => {
                   this.info(`Importing APIs
-                      ${bName} ${bVersion} ${osName} ${osVersion}`);
-                  return this.apiImporter.import(bName, bVersion, osName,
-                                                 osVersion, catalog);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
+                  return this.apiImporter.import(
+                      releaseInfo.browser.name, releaseInfo.browser.version,
+                      releaseInfo.platform.name, releaseInfo.platform.version,
+                      catalog);
                 }).then(() => {
                   this.info(`APIs imported for
-                                 ${bName} ${bVersion} ${osName} ${osVersion}`);
+                      ${releaseInfo.browser.name} ${releaseInfo.browser.version}
+                      ${releaseInfo.platform.name}
+                      ${releaseInfo.platform.version}`);
                 }));
           }
         }
         return Promise.all(promises);
       },
     },
+    function getReleaseInfo(filename) {
+      const json = JSON.parse(fs.readFileSync(
```

Comment:
This means that each imported file will be loaded into memory and parsed twice, right?

Reply:
Sad, but true. We temporarily extract the file to get release info, then throw it away. Later, worker processes extract it for API extraction.

Reply:
Would it be unreasonable for the API extractor to also return the release info? It already has the information (file content or file name).

Reply:
That seems reasonable. Just a bigger change because right now its return value is an API listing (with no metadata).